pandas/core/frame.py

```diff
@@ -5919,6 +5919,9 @@ def dropna(
 
         agg_obj = self
         if subset is not None:
+            if isinstance(subset, str):
```

Comment:
Worth adding a test to make sure reasonable behavior if `subset` passed as a string, but is not present in relevant axis?

Reply:
Added a test for this. We check for `KeyError` when given column is not in axis.